processor/src/debug.rs

```diff
@@ -55,14 +56,23 @@ pub struct VmStateIterator {
     clk: u32,
     asmop_idx: usize,
     forward: bool,
+    trace_len_summary: TraceLenSummary,
 }
 
 impl VmStateIterator {
     pub(super) fn new<H>(process: Process<H>, result: Result<StackOutputs, ExecutionError>) -> Self
     where
         H: Host,
     {
-        let (system, decoder, stack, _, chiplets, _) = process.into_parts();
+        let (system, decoder, stack, mut range, chiplets, _) = process.into_parts();
+
+        let clk = system.clk();
+        let range_table_len = range.get_number_range_checker_rows();
+        chiplets.append_range_checks(&mut range);
+
+        let trace_len_summary =
+            TraceLenSummary::new(clk as usize, range_table_len, ChipletsLengths::new(&chiplets));
```

Comment:
Let's create a separate function to encapsulate this logic. Maybe something like `build_trace_len_summary(system, range, chipletes)`.